src/gate_detect/test/gate-detection-test.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Created By: Cameron Newton
+ * Created On: May 5th, 2018
+ * Description: Test functionality of GateDetection
+ */
+
+#include "GateDetection.h"
+#include "testUtilities/TestUtils.h"
+#include <gtest/gtest.h>
+
+using namespace cv;
+using namespace std;
+
+TEST(TestSuite, testCase1) {
+    Mat image;
+
+    image = imread("testImages/gateFrontOrange.png",
+                   CV_LOAD_IMAGE_COLOR); // Read the file
+
+    if (image.empty()) // Check for invalid input
+    {
+        cout << "Could not open or find the image" << std::endl;
+        FAIL();
+    }
+
+    GateDetection gateDetection;
+
+    std::vector<float> floater = gateDetection.initialize(image);
+
+    // Uncomment to view gate drawn over image
+    // TestUtils::DisplayGateDetected(image,floater);
```

Comment:
We should probably put these display statements inside a conditional and just make it constant at the top ie:
```
if (DRAW_DEBUG_IMAGES){
    TestUtils::DisplayGateDetected(image,floater);
}
````

Reply:
We don't actually seem to be checking anything in this test? Same goes for a lot of the tests here?

Reply:
They're really visual tests, if you uncomment the TestUtils you can see the results. I don't really know how to write assert_true_if tests for this kind of functionality. I was going to do a more in depth test suite but it seems like a time sink